code/game/objects/effects/effect_system.dm

```diff
@@ -42,8 +42,7 @@ would spawn and follow the beaker, even if it is carried or thrown.
 /obj/effect/proc/delete()
 	loc = null
 	if(reagents)
-		reagents.my_atom = null
-		reagents.delete()
+		qdel(reagents)

```

Comment:
I was actually planning on working on the effects system after this PR got merged in--and it's a bit redundant since reagents get deleted later on up the Destory() stack.